### PR TITLE
Simplify scm/git data flow

### LINKED ISSUE #5295: simplify scm/git data flow
`ScmService` should be a single source of the whole scm state, git extension should populate it and provide git specific view instead of duplicating this state partially and trying to sync it.

For right now repositories state is duplicated several times:
- `GitContribution.dirtyResporitories` duplicating `GitRepositoryProvider.allRepositories` which duplicating `ScmService.repositories`
- `GitRepositoryProvider.selectedRepository` duplicating `ScmService.selectedRepository`

`GitRepositoryProvider` should be a git specific facade to `ScmService`. Handling of `ScmProvider` should be moved to `GitRepositoryProvider` instead of keeping allRepositories and `GitContribution.dirtyResporitories`.

### PR DESCRIPTION
**Pull request by vinokurig**
fixes #5295

### CODE REVIEW

**Comment by akosyakov** [comment]
- `GitRepositoryProvider` apis should be preserved, it is used a lot
- SCM providers which are not contributed by git extension should be ignores, i.e. if not git extension provider is selected then it should be considered as not selected

File: packages/git/src/browser/git-repository-provider.ts [hunk 1]
```diff
@@ -71,9 +73,7 @@ export class GitRepositoryProvider {
             this.lazyRefresh();
         });
 
-        this._selectedRepository = await this.storageService.getData<Repository | undefined>(this.selectedRepoStorageKey);
```
Comment (akosyakov) [inline packages/git/src/browser/git-repository-provider.ts]: Why did you remove it? It is done that git does not empty UI before actual state is loaded.

Comment (vinokurig) [reply packages/git/src/browser/git-repository-provider.ts]: `ScmRepository` is a class, so it won't be serialized in the `StorageService`

**Comment by vinokurig** [comment]
@akosyakov 
> GitRepositoryProvider apis should be preserved, it is used a lot by extenders
SCM providers which are not contributed by git extension should be ignored, i.e. if not git extension provider is selected then it should be considered as not selected

Done

File: packages/git/src/browser/git-repository-provider.ts (hunk 1 above)
Comment (akosyakov) [reply packages/git/src/browser/git-repository-provider.ts]: Could we jsonize scmrepositor data and recover classes on reload from such data?

Comment (vinokurig) [reply packages/git/src/browser/git-repository-provider.ts]: `ScmRepository` consumes handlers: https://github.com/theia-ide/theia/blob/71f277be8505b8e405e945854e07edc6c12f1200/packages/scm/src/browser/scm-service.ts#L201 that are not serialized

**Comment by akosyakov** [comment]
@vinokurig I will merge your commits in a new PR and fix new issues (https://github.com/theia-ide/theia/issues/5374, https://github.com/theia-ide/theia/issues/5375) on the top.

File: packages/git/src/browser/git-repository-provider.ts (hunk 1 above)
Comment (akosyakov) [reply packages/git/src/browser/git-repository-provider.ts]: ok, will have a look what can be done
